### PR TITLE
Publicize: Add jetpack_publicize_options

### LINKED ISSUE #565: Publicize: add filter to disable enhanced custom stories / embedded images
Publicize changes the look of your Facebook updates if you publish Image Posts, or posts including very little text and only an image. When you do so, the image is uploaded to Facebook, and the Publicized post looks like an image you'd have directly uploaded to Facebook.

While this is great since it generates more Likes and Comments on Facebook, it doesn't bring as much traffic to your site as a regular shared link.
Some people also don't feel comfortable having their images uploaded to Facebook.

Would it be possible to add a filter allowing blog owners to use the default link format in their Facebook Publicized posts?

Suggested in 1844983-t

### PR DESCRIPTION
**Pull request by kraftbj**
Fixes #565

Still working on testing it; posting for easier discussion.

#### Changes proposed in this Pull Request:
* Syncs the output of a new `jetpack_publicize_options` filter to WP.com to allow locally set options to influence Publicize.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* pbAAop-lT-p2

#### Testing instructions:
* Add this to the Jetpack site:
`add_filter( 'jetpack_publicize_options', function() { return array( 'attach_media' => false ); } );`
* Setup Publicize with Twitter on the Jetpack site.
* Publish. See the tweet using the Twitter card instead of embedding the image.

#### Proposed changelog entry for your changes:
* Publicize: Allow site owners to set a filter to disable embedding media directly into the social media post.

[Edited description as WP.com patch required for this has been committed.]

### CODE REVIEW

**Comment by kraftbj** [comment]
Force-pushed to knock GitHub into trying everything again after their outage this morning.

File: packages/sync/src/class-defaults.php [hunk 1]
```diff
@@ -293,6 +293,7 @@ public static function get_constants_whitelist() {
 		'available_jetpack_blocks'         => array( 'Jetpack_Gutenberg', 'get_availability' ), // Includes both Gutenberg blocks *and* plugins.
 		'paused_themes'                    => array( 'Automattic\\Jetpack\\Sync\\Functions', 'get_paused_themes' ),
 		'paused_plugins'                   => array( 'Automattic\\Jetpack\\Sync\\Functions', 'get_paused_plugins' ),
+		'publicize_options'                => array( 'Automattic\\Jetpack\\Sync\\FUnctions', 'get_publicize_options' ),
```
Comment (WPprodigy) [inline packages/sync/src/class-defaults.php:296]:
```suggestion
		'publicize_options'                => array( 'Automattic\\Jetpack\\Sync\\Fhnctions', 'get_publicize_options' ),
```

Comment (WPprodigy) [inline packages/sync/src/class-defaults.php:296]:
```suggestion
		'publicize_options'                => array( 'Automattic\\Jetpack\\Sync\\Functions', 'get_publicize_options' ),
```

**Comment by kraftbj** [comment]
2020-04-27 at 11 17 AM

**Comment by kraftbj** [comment]
r206515-wpcom

**Pull request merged**